File: BrasilBurger_Java/src/main/java/com/brasilburger/models/Menu.java
package com.brasilburger.models;

import java.util.List;

public class Menu {
    private int id;
    private String nom;
    private String image;
    private List<Burger> burgers;
    private List<Complement> complements;
    private boolean archive;

    public Menu() {}

    public Menu(int id, String nom, String image, List<Burger> burgers, List<Complement> complements, boolean archive) {
        this.id = id;
        this.nom = nom;
        this.image = image;
        this.burgers = burgers;
        this.complements = complements;
        this.archive = archive;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }

    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }

    public List<Burger> getBurgers() { return burgers; }
    public void setBurgers(List<Burger> burgers) { this.burgers = burgers; }

    public List<Complement> getComplements() { return complements; }
    public void setComplements(List<Complement> complements) { this.complements = complements; }

    public boolean isArchive() { return archive; }
    public void setArchive(boolean archive) { this.archive = archive; }

    @Override
    public String toString() {
        return nom + " (" + (burgers != null ? burgers.size() : 0) + " burgers)";
    }
}

File: BrasilBurger_Symfony/src/Entity/Menu.php
<?php

namespace App\Entity;

use App\Repository\MenuRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: MenuRepository::class)]
class Menu
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: 'Le nom du menu est obligatoire')]
    private ?string $nom = null;

    #[ORM\Column(length: 500, nullable: true)]
    private ?string $image = null;

    #[ORM\Column]
    private ?bool $archive = false;

    #[ORM\OneToMany(mappedBy: 'menu', targetEntity: MenuItem::class, cascade: ['persist', 'remove'])]
    private Collection $menuItems;

    #[ORM\OneToMany(mappedBy: 'menu', targetEntity: CommandeItem::class, cascade: ['persist', 'remove'])]
    private Collection $commandeItems;

    public function __construct()
    {
        $this->menuItems = new ArrayCollection();
        $this->commandeItems = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): static
    {
        $this->image = $image;

        return $this;
    }

    public function isArchive(): ?bool
    {
        return $this->archive;
    }

    public function setArchive(bool $archive): static
    {
        $this->archive = $archive;

        return $this;
    }

    /**
     * @return Collection<int, MenuItem>
     */
    public function getMenuItems(): Collection
    {
        return $this->menuItems;
    }

    public function addMenuItem(MenuItem $menuItem): static
    {
        if (!$this->menuItems->contains($menuItem)) {
            $this->menuItems->add($menuItem);
            $menuItem->setMenu($this);
        }

        return $this;
    }

    public function removeMenuItem(MenuItem $menuItem): static
    {
        if ($this->menuItems->removeElement($menuItem)) {
            // set the owning side to null (unless already changed)
            if ($menuItem->getMenu() === $this) {
                $menuItem->setMenu(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection<int, CommandeItem>
     */
    public function getCommandeItems(): Collection
    {
        return $this->commandeItems;
    }

    public function addCommandeItem(CommandeItem $commandeItem): static
    {
        if (!$this->commandeItems->contains($commandeItem)) {
            $this->commandeItems->add($commandeItem);
            $commandeItem->setMenu($this);
        }

        return $this;
    }

    public function removeCommandeItem(CommandeItem $commandeItem): static
    {
        if ($this->commandeItems->removeElement($commandeItem)) {
            // set the owning side to null (unless already changed)
            if ($commandeItem->getMenu() === $this) {
                $commandeItem->setMenu(null);
            }
        }

        return $this;
    }

    public function getPrix(): float
    {
        $prix = 0;
        foreach ($this->menuItems as $item) {
            if ($item->getBurger()) {
                $prix += $item->getBurger()->getPrix();
            }
            if ($item->getComplement()) {
                $prix += $item->getComplement()->getPrix();
            }
        }
        return $prix;
    }
}
